rednotebook/util/markup.py

```diff
@@ -226,6 +226,26 @@ def _get_config(target, options):
         # {{red text|color:red}} -> <span style="color:red">red text</span>
         config['postproc'].append([COLOR_ESCAPED, r'<span style="color:\2">\1</span>'])
 
+        # Entry references
+        if is_exporting_to_file:
+            # When exporting we need to remove entry reference links because
+            # they are only valid within our app.
+            config['preproc'].append([r'\[(?P<name>.+)\s+(?P<date>\d{4}-\d{2}-\d{2})\s*\]',
+                                      r'\g<name> (\g<date>)'])
+        else:
+            # txt2tag will generate links to the named entry references (e.g. "[Some text 2019-10-15]").
+            # Here we just need to add our internal URI schema.
+            config['postproc'].append([r'href="(?P<date>\d{4}-\d{2}-\d{2})"',
+                                       r'href="' + urls.INTERNAL_URI_SCHEMA + r':\g<date>"'])
+
+            # For simple notation (e.g. "2019-10-15") we need to generate full link code being careful
+            # not to override already existing named links.
+            # Here we look for dates NOT preceded by our internal URI schema.
+            # NOTE: When an entry reference name contains a date, this rule may generate
+            # nested links, e.g: "<a...><a...>...</a></a>", but this should not affect the user.
+            config['postproc'].append([r'(?<!' + urls.INTERNAL_URI_SCHEMA + r':)(?P<date>\d{4}-\d{2}-\d{2})',
+                                       r'<a href="' + urls.INTERNAL_URI_SCHEMA + r':\g<date>">\g<date></a>'])
+
```

Comment:
Unfortunately there is a catch - after preproc, named references will end up like that:
`[Named reference [2019-08-01]]`
Which in turn will mangle URLs to:
`<a href="notebook:2019-08-01">Named reference [2019-08-01</a>]`

To mitigate that we could check if the date is already part of named entry reference... but it will be hell either way :-/

Reply:
Well... while working on anchors I think I have found a cleaner, `preproc` solution so it is done too :-)